common/pipeline_storage/src/test/scala/uk/ac/wellcome/pipeline_storage/PipelineStorageStreamTest.scala

```diff
@@ -260,6 +261,37 @@ class PipelineStorageStreamTest
     }
   }
 
+  it(
+    "does not delete the message if the processFunction throws an exception not wrapped in a future") {
+    val index = createIndex
+    val document = SampleDocument(1, createCanonicalId, randomAlphanumeric())
+
+    val sender = new MemoryMessageSender
+
+    withLocalSqsQueuePair(visibilityTimeout = 10) {
+      case QueuePair(queue, dlq) =>
+        withPipelineStream(
+          queue = queue,
+          indexer = indexer(index),
+          sender = sender) { pipelineStream =>
+          sendNotificationToSQS(
+            queue = queue,
+            message = document
+          )
+          pipelineStream.foreach(
+            "test stream",
+            _ => throw new Exception("Boom!"))
+
+          eventually(Timeout(Span(90, Seconds))) {
```

Comment:
Do we really need to wait 90 seconds for this test to fail? Couldn't we crank down the visibility timeout to make it go faster?

Reply:
Ah sorry that's a leftover. It wasn't failing so I cranked up the timeout to see if it would fail waiting longer. I'll take it out